TypeExtender/TypeExtender/TypeExtender.cs

```diff
@@ -173,6 +173,47 @@ public class TypeExtender {
                 addProperty(properties.ElementAt(index).Replace(" ","_"), types.ElementAt(index), null, null, allReadOnly);
             }
         }
+        
```

Comment:
Most of the public methods have a Generic overload... 

Please implement a generic overload for the AddField function.